src/backend/distributed/utils/connection_cache.c

```diff
@@ -30,8 +30,8 @@
 #include "utils/palloc.h"
 
 
-/* state needed to prevent new connections during modifying transactions */
-bool IsModifyingTransaction = false;
+/* state needed to keep track of operations used during a transaction */
+XactModificationType XactModification = XACT_MODIFICATION_NONE;
```

Comment:
I was going to suggest `XactModificationType`, but that's the type name. Bah!

I'll do `Level`. There isn't anything intrinsically "greater" about schema modifications over data ones, but one could argue the locks are often stronger, so…? I thought about `Level` myself here independently, so let's go with that.